### ISSUE TITLE
rsync: [sender] change_dir failed

### ISSUE THREAD

**Issue by farcaller** [nix-community/NixOS-WSL#164]
## Bug description
On a cold wsl startup:

```
Copying /usr/share/applications
Copying /usr/share/icons
rsync: [sender] change_dir "/nix/store/4c5067qszqva944xynbff1xss8b9z0h2-nixos-system-wsl-22.11.20221110.5f588eb/sw/share/icons" failed: No such file or directory (2)
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1327) [sender=3.2.6]
Activation script snippet 'copy-launchers' failed (23)
setting up /etc...
setting up /bin...

[process exited with code 1 (0x00000001)]
```

Reproduced at e9e8d01c5a2f57d330d42c9a772cf67d89b43650

## To Reproduce
```
wsl --shuutdown
(wait 8 seconds)
wsl
```

**Expected behavior**
wsl starts up with no issues

## Logs
See above

**Comment by macharro** [comment]
I had the exact same error after upgrading to 22.11.

I did work around the error by disableing the `copy-launchers` activation script by change the `wsl.startMenuLaunchers` to false in your configuration.nix file.

I traced the error this line:
https://github.com/nix-community/NixOS-WSL/blob/e9e8d01c5a2f57d330d42c9a772cf67d89b43650/modules/wsl-distro.nix#L87

**Comment by nzbr** [comment]
Can one of you attach your `configuration.nix`? I'm not able to reproduce the problem myself right now

**Comment by LGUG2Z** [comment]
I had this problem on the latest official release, but then I grabbed the latest artifacts from the most recent successful GitHub Actions run and it was no longer present.

**Comment by SuperSandro2000** [comment]
@nzbr I think this is happening if your configuration has no icons at all and the ``/run/current-system//sw/share/icons`` does not exist. Didn't test though.

**Comment by zeratax** [comment]
I have this too. But this only happens if i directly open a NixOS wsl. if I in powershell just execute `wsl` the errors still appear, but i still enter NixOS. Afterwards I can open NixOS in any way and it just works.

my configuration:
```nix
{ lib, pkgs, config, modulesPath, ... }:

with lib;
let
  nixos-wsl = import ./nixos-wsl;
in
{
  imports = [
    "${modulesPath}/profiles/minimal.nix"
    ./cachix.nix
    nixos-wsl.nixosModules.wsl
  ];

  wsl = {
    enable = true;
    automountPath = "/mnt";
    defaultUser = "nixos";
    startMenuLaunchers = true;

    # Enable native Docker support
    # docker-native.enable = true;

    # Enable integration with Docker Desktop (needs to be installed)
    # docker-desktop.enable = true;

  };

  # Enable nix flakes
  nix.package = pkgs.nixFlakes;
  nix.extraOptions = ''
    experimental-features = nix-command flakes
  '';

  system.stateVersion = "22.05";
}
```

**Comment by nialov** [comment]
Happening here as well with update to `NixOS-wsl` rev 2e1cbbb565c6ad8ca7e5f17c743bb127b22d48f7 and `nixpkgs` rev 5efc8ca954272c4376ac929f4c5ffefcc20551d5.

I suppose there are multiple ways to still access the distro but personally I rolled back with this workflow:

Enter wsl (`-u root` probably does not matter) without the activation script error:

~~~powershell
wsl -d NixOS -u root -- bash --noprofile --norc
~~~

~~~bash
# List generations and their ids
sudo nix-env --list-generations --profile /nix/var/nix/profiles/system
# Switch to a specific one
sudo nix-env --profile /nix/var/nix/profiles/system --switch-generation '<id>'
~~~

Not a solution but maybe helpful to someone...

**Comment by ErikMcClure** [comment]
I also encountered this error when trying to update to 22.11, but I was unable to save the distro using nialov's method above - it was unrecoverable, and I simply copied the files out to a fresh NixOS distro. It is not possible to fix this error by creating an `icons` directory in the place that it asks for because it seems to symlink to a read-only file-system (some WSL location?). I simply turned off the `startmenulaunchers` option, but it seems like [this script](https://github.com/nix-community/NixOS-WSL/blob/e9e8d01c5a2f57d330d42c9a772cf67d89b43650/modules/wsl-distro.nix#L84) should be doing a check to see if the icons and applications folders it wants to sync actually exist before doing the copy.
